ddl/ddl.go

```diff
@@ -506,6 +508,47 @@ func (d *ddl) addColumn(schema model.CIStr, tbl table.Table, spec *AlterSpecific
 	return errors.Trace(err)
 }
 
+func updateDefaultValue(ctx context.Context, t *tables.Table, col *column.Col) error {
+	txn, err := ctx.GetTxn(false)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	it, err := txn.Seek([]byte(t.FirstKey()), nil)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer it.Close()
+
+	prefix := t.KeyPrefix()
+	for it.Valid() && strings.HasPrefix(it.Key(), prefix) {
+		handle, err0 := util.DecodeHandleFromRowKey(it.Key())
+		if err0 != nil {
+			return errors.Trace(err0)
+		}
+		k := t.RecordKey(handle, col)
+		colID, err0 := tables.ColumnID(k)
```

Comment:
yep.
here `colID == col.ID` is always true.  Because the key is produced by `RecordKey(handle, col)`。